File: port_Go/parser/python_parser_base.go
package parser

import "github.com/antlr4-go/antlr/v4"

type PythonParserBase struct {
	*antlr.BaseParser
}

func NewPythonParserBase(input antlr.TokenStream) *PythonParserBase {
	ppb := new(PythonParserBase)
	ppb.BaseParser = antlr.NewBaseParser(input)
	return ppb
}

func (prs *PythonParserBase) isEqualToCurrentTokenText(tokenText string) bool {
	return prs.GetCurrentToken().GetText() == tokenText
}

func (prs *PythonParserBase) isnotEqualToCurrentTokenText(tokenText string) bool {
	return !prs.isEqualToCurrentTokenText(tokenText)
}

File: port_Dart/bin/PythonParserBase.dart
import 'package:antlr4/antlr4.dart';

abstract class PythonParserBase extends Parser {
  PythonParserBase(TokenStream input) : super(input);

  // https://docs.python.org/3/reference/lexical_analysis.html#soft-keywords
  bool isEqualToCurrentTokenText(String tokenText) {
    return this.currentToken.text == tokenText;
  }

  bool isnotEqualToCurrentTokenText(String tokenText) {
    return !this.isEqualToCurrentTokenText(tokenText); // for compatibility with the Python 'not' logical operator
  }
}
